### PR TITLE
Version bumped TRINITY, added nf-tests and stub

### PR DESCRIPTION
**Pull request by GallVp**
## PR checklist

Closes #XXX 

- [x] Version bumped TRINITY, added nf-tests and stub
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/trinity/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.versions).match("versions") },
+                { assert path(process.out.log[0][1]).text.contains("All commands completed successfully.") },
+                { assert process.out.transcript_fasta != null },
```
[the remark this answers is not in the document]
Comment (GallVp) [reply modules/nf-core/trinity/tests/main.nf.test]: Thank you for the feedback @CarsonJM 

I have created snapshots for the fasta files.

[the remark this answers is not in the document]
Comment (GallVp) [reply modules/nf-core/trinity/tests/main.nf.test]: Unfortunately the md5 for the fasta was unstable and I had to revert back.

File: modules/nf-core/trinity/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.versions).match("versions") },
+                { assert process.out.transcript_fasta != null },
```
[the remark this answers is not in the document]
Comment (GallVp) [reply modules/nf-core/trinity/tests/main.nf.test]: I have added a stub which lists file names.

**Comment by GallVp** [comment]
Hi @CarsonJM 

I think there is an issue with the nf-test. I don't think it is behaving as expected. Btw, the following versions snapshot was to ensure that, in future, if the versions snippet is updated in `script` without updating it in the `stub`, a common snapshot will capture that bug.

```groovy
{ assert snapshot(process.out.versions).match("versions") },
```

**Comment by CarsonJM** [comment]
@GallVp Thanks for letting me know! I had the snapshot for `transcript_fasta` working reproducibly in Gitpod by using only one thread, but for some reason it did translate to Codespaces or Github Actions. I am about to push a change fixing the CI using a `.linesGzip.toString().contains` for the `transcript_fasta` so there is at least some check of file content and I will add back in the `.match("versions")` as well. 

Sorry for the lack of communication before I made these changes, I just wanted to try and troubleshoot the md5 issue to see if there was any chance of getting it to work!

**Review by CarsonJM — APPROVED** [review]
Nice work! Thank you for your patience!

**Comment by GallVp** [comment]
Thank you @CarsonJM for your help.

**Pull request merged**